rustls/src/webpki/client_verifier.rs

```diff
@@ -142,12 +142,7 @@ impl ClientCertVerifierBuilder {
             .supported_algs
             .ok_or(VerifierBuilderError::NoSupportedAlgorithms)?;
 
-        let root_hint_subjects = self
-            .roots
-            .roots
-            .iter()
-            .map(|ta| DistinguishedName::in_sequence(ta.subject.as_ref()))
-            .collect();
+        let root_hint_subjects = self.roots.subjects();
```

Comment:
Consider going back to the way it was, inlining the initialization of the `root_hint_subjects` into the `WebPkiClientVerifier::new()` call?

Reply:
I don't think that will work - the `self.roots` argument comes first in the `WebPkiClientVerifier::new` argument list, and using `self.roots.subjects()` as the arg afterwards is trying to use a moved value. We have to pull it out before moving `roots`.

Reply:
Ultimately we end up in the state you want by the end of the commits once the `root_hint_subjects` are stored in the builder separately from `roots`: https://github.com/cpu/rustls/blob/3d0960b3a3f7077d9d6e23cc56d8825134b1ba93/rustls/src/webpki/client_verifier.rs#L178-L181